src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/FhirJsonOutputFormatter.cs

```diff
@@ -70,7 +72,10 @@ public override async Task WriteResponseBodyAsync(OutputFormatterWriteContext co
                     jsonWriter.Formatting = Formatting.Indented;
                 }
 
-                _fhirJsonSerializer.Serialize((Resource)context.Object, jsonWriter, context.HttpContext.GetSummaryType(_logger));
+                _fhirJsonSerializer.Serialize(
+                    (Resource)context.Object,
+                    jsonWriter,
+                    (context.HttpContext.Response.StatusCode == (int)HttpStatusCode.OK) ? context.HttpContext.GetSummaryType(_logger) : SummaryType.False);
```

Comment:
I feel like this logic should be resolved in `GetSummaryType`. Perhaps in this case if we're already validating the logic we change `GetSummaryType` to `TryParse` of the enum and don't throw

Reply:
I agree to the logic to get resolved inside GetSummaryType if that's meant only for this specific use case (which is actually the case right now). I think handling the logic outside allows GetSummaryType to be used in other scenarios as well (which expect exception in case of invalid value). In last discussion, Jack was inclined towards latter (current change). Let's discuss it together when he's back.

Reply:
Moved the check inside GetSummaryType.